core/src/main/java/org/dcache/nfs/v4/OperationOPEN.java

```diff
@@ -206,6 +206,7 @@ public void process(CompoundContext context, nfs_resop4 result) throws ChimeraNF
                 }
 
                 context.currentInode(inode);
+                context.getFs().open(inode, this.getAccessMode(_args.opopen.share_access), result.opopen.resok4.stateid);
```

Comment:
you will mess all opens by file handle, which is quite common, when client already got hanlde by  previous readdir, lookup or open.